torchaudio/models/wav2vec2/components.py

```diff
@@ -397,6 +401,29 @@ def forward(
 
         return x
 
+    def get_intermediate_outputs(
+            self,
+            x: Tensor,
+            attention_mask: Optional[Tensor] = None,
+            indices: Optional[List[int]] = None,
```

Comment:
The default value of indices can be the last layer? But it looks difficult to set such value for models of different sizes..

Reply:
The current default is to return the all layers.

```python
model = Wav2Vec2Model(...)
features, _ = model.extract_features(waveform, length, indices=None)

# features is a list of tensors, which consists of outputs from all the intermediate layers in transformer block.
```

I am open to change it to other formats, and here are the design criteria I would like to keep,
1. Since it is computationally heavy, we would like the interface to be flexible enough to extract features from multiple layers at the same time.
2. Once the output from last layer of the desired layers is extracted, stop the computation there. (no need to go through the rest of transformer layers.)

`fairseq` returns dictionary with `x` represents the last layer output, and `layer_resutls` contain the list of intermediate outputs.

The last layer output can be fetched like,

```python
features, _ = model.extract_features(waveform, length, indices=None)
last_layers = features[-1]
```

What do you think?

Reply:
I agree with the design criteria. To save the computational time, we may add a parameter like ``last_only=True`` which only extract the last layer's output by default. The logic can be
```
if indices is None:
    if last_only is True:
        # return last layer output
    else:
        # return all layer outputs
else:
    # return desired layer outputs

```